### PR TITLE
TST: pytest deprecation warnings GH17197

### LINKED ISSUE #17197: TST: pytest deprecation warnings
xref https://travis-ci.org/pandas-dev/pandas/jobs/262021774

lots of these, looks like pytest 3.2 triggers these.

```
/home/travis/miniconda3/envs/pandas/lib/python3.6/site-packages/_pytest/mark.py:66: RemovedInPytest4Warning: Applying marks directly to parameters is deprecated, please use pytest.param(..., marks=...) instead.
For more details, see: https://docs.pytest.org/en/latest/parametrize.html
  warnings.warn(MARK_PARAMETERSET_UNPACKING)
```

### PR DESCRIPTION
**Pull request by swyoon**
Test parameters with marks are updated according to the updated API of
Pytest.
https://docs.pytest.org/en/latest/changelog.html#pytest-3-2-0-2017-07-30
https://docs.pytest.org/en/latest/parametrize.html

- [x] closes #17197 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.21.0.txt [hunk 1]
```diff
@@ -386,3 +386,4 @@ Other
 - Bug in :func:`eval` where the ``inplace`` parameter was being incorrectly handled (:issue:`16732`)
 - Bug in ``.isin()`` in which checking membership in empty ``Series`` objects raised an error (:issue:`16991`)
 - Bug in :func:`unique` where checking a tuple of strings raised a ``TypeError`` (:issue:`17108`)
+- Fixes RemovedInPytest4Warning by using pytest.param(..., marks=...) (:issue:`17197`)
```
Comment (gfyoung) [inline doc/source/whatsnew/v0.21.0.txt]: You don't need a `whatsnew` for this type of change.

**Review by gfyoung — APPROVED** [review]
Build looks clean to me.  @jreback

**Comment by jreback** [comment]
can you figure the earliest version of pytest that we should allow?

and want to add this to the ci instructions and note in the contrib docs (we aren't going to test with an older version though).

**Comment by jreback** [comment]
I didn't realize that this deprecation in pytest has a much more verbose syntax.

**Comment by swyoon** [comment]
Thanks for reviewing the PR.

`pytest.param` was introduced in pytest version 3.1.0, and therefore, the new test code only work with pytest version 3.1.0 or later. Errors occur when pytest version 3.0.7 (the latest release before 3.1.0) is used.
https://docs.pytest.org/en/latest/changelog.html#id59

will add notes to `pandas/doc/contributing.md`.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jreback** [comment]
so need to update with ``pytest>=3.1.0``
```
appveyor.yml: - cmd: conda create -n pandas python=%PYTHON_VERSION% cython pytest pytest-xdist
ci/install_circle.sh:time conda install -n pandas pytest || exit 1
ci/install_travis.sh:time conda install -n pandas pytest
ci/requirements_all.txt:pytest
ci/requirements_dev.txt:pytest
````

**Comment by swyoon** [comment]
will update `appveyor.yml`.

btw, it is strange that trevis build breaks after me rebasing onto the current master. `TestClipboard.test_round_trip_frame_sep` and `TestClipboard.test_round_trip_frame` failed, but it seems there's nothing much to do with this PR. Any ideas?

**Comment by jreback** [comment]
> btw, it is strange that trevis build breaks after me rebasing onto the current master. TestClipboard.test_round_trip_frame_sep and TestClipboard.test_round_trip_frame failed, but it seems there's nothing much to do with this PR. Any ideas?

these occasionally fail, ignore

**Comment by swyoon** [comment]
`appveyor.yml` and other requirements files are updated, and the builds are green.

File: doc/source/contributing.rst [hunk 2]
```diff
@@ -709,6 +713,10 @@ A test run of this yields
 
 Tests that we have ``parametrized`` are now accessible via the test name, for example we could run these with ``-k int8`` to sub-select *only* those tests which match ``int8``.
 
+.. note::
+
```
Comment (jreback) [inline doc/source/contributing.rst]: can you show how to mark a test as skip/xfail (rather than this note)

**Comment by jreback** [comment]
small comment on docs. otherwise lgtm. ping on green.

**Comment by swyoon** [comment]
added and updated `pytest` usage examples. hope it makes the documentation clear. builds are all green.

**Comment by jorisvandenbossche** [comment]
@swyoon Thanks for the nice PR!

**Comment by swyoon** [comment]
thank you all for kind reviewing. will hope to contribute to pandas again.

**Pull request merged**